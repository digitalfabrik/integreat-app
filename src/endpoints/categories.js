// @flow

import CategoryModel from '../models/CategoryModel'
import CategoriesMapModel from '../models/CategoriesMapModel'
import EndpointBuilder from '../EndpointBuilder'
import moment from 'moment'
import type { JsonCategoryType } from '../types'
import mapAvailableLanguages from '../mapAvailableLanguages'
import normalizePath from '../normalizePath'
import Endpoint from '../Endpoint'
import sanitizeHtml from 'sanitize-html-react'

const CATEGORIES_ENDPOINT_NAME = 'categories'

type ParamsType = { city: string, language: string }

const endpoint: Endpoint<ParamsType, CategoriesMapModel> = new EndpointBuilder(CATEGORIES_ENDPOINT_NAME)
  .withParamsToUrlMapper((apiUrl: string, params: ParamsType): string =>
    `${apiUrl}/${params.city}/${params.language}/wp-json/extensions/v3/pages`
  )
  .withMapper((json: Array<JsonCategoryType>, params: ParamsType): CategoriesMapModel => {
    const basePath = `/${params.city}/${params.language}`

    const categories = json
      .map(category => {
        return new CategoryModel({
          id: category.id,
          path: normalizePath(category.path),
          title: category.title,
          content: sanitizeHtml(category.content, {
            allowedSchemes: ['http', 'https', 'data', 'tel', 'mailto'],
            allowedTags: false,
            allowedAttributes: false
          }),
          thumbnail: category.thumbnail,
          order: category.order,
          availableLanguages: mapAvailableLanguages(category.available_languages),
          parentPath: normalizePath(category.parent.path || basePath),
          lastUpdate: moment(category.modified_gmt)
        })
      })

    categories.push(new CategoryModel({
      id: 0,
      path: basePath,
      title: params.city,
      parentPath: '',
      content: '',
      thumbnail: '',
      order: -1,
      availableLanguages: new Map(),
      lastUpdate: moment(0)
    }))

    return new CategoriesMapModel(categories)
  })
  .build()

export default endpoint
