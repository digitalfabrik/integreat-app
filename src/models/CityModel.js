// @flow

class CityModel {
  _name: string
  _code: string
  _live: boolean
  _eventsEnabled: boolean
  _extrasEnabled: boolean
  _sortingName: string
  _prefix: ?string
  _latitude: number | null
  _longitude: number | null
  _aliases: { [alias: string]: {| longitude: number, latitude: number |} } | null

  constructor (params: {|
    name: string, code: string, live: boolean, eventsEnabled: boolean, extrasEnabled: boolean,
    sortingName: string, prefix: ?string, latitude: number | null, longitude: number | null,
    aliases: { [alias: string]: {| longitude: number, latitude: number |} } | null
  |}) {
    this._name = params.name
    this._code = params.code
    this._live = params.live
    this._eventsEnabled = params.eventsEnabled
    this._extrasEnabled = params.extrasEnabled
    this._sortingName = params.sortingName
    this._prefix = params.prefix
    this._latitude = params.latitude
    this._longitude = params.longitude
    this._aliases = params.aliases
  }

  get live (): boolean {
    return this._live
  }

  get name (): string {
    return this._name
  }

  get code (): string {
    return this._code
  }

  get sortingName (): string {
    return this._sortingName
  }

  get eventsEnabled (): boolean {
    return this._eventsEnabled
  }

  get extrasEnabled (): boolean {
    return this._extrasEnabled
  }

  get sortCategory (): string {
    return this._sortingName.charAt(0)
  }

  get prefix (): ?string {
    return this._prefix
  }

  get longitude (): number | null {
    return this._longitude
  }

  get latitude (): number | null {
    return this._latitude
  }

  get aliases (): { [alias: string]: {| longitude: number, latitude: number |} } | null {
    return this._aliases
  }

  static findCityName (cities: Array<CityModel>, code: string): string {
    const city = cities.find(city => city.code === code)
    return city ? city.name : code
  }
}

export default CityModel
