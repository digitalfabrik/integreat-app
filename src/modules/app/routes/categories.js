// @flow

import categoriesEndpoint from '../../endpoint/endpoints/categories'
import { createAction } from 'redux-actions'

import type { Dispatch, GetState } from 'redux-first-router/dist/flow-types'

export const CATEGORIES_ROUTE = 'CATEGORIES'

export const goToCategories = (city: string, language: string, categoryPath: ?string) =>
  createAction(CATEGORIES_ROUTE)({city, language, categoryPath})

/**
 * CategoriesRoute, matches /augsburg/de*
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
export const categoriesRoute = {
  path: '/:city/:language/:categoryPath*',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()
    const {city, language} = state.location.payload

    await categoriesEndpoint.loadData(dispatch, state.categories, {city, language})
  }
}
