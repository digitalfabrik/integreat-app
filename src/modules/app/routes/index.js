// @flow

import categoriesRoute from './categories'
import eventsRoute from './events'
import { EXTRAS_ROUTE, extrasRoute, getExtrasPath, renderExtrasPage } from './extras'
import { DISCLAIMER_ROUTE, disclaimerRoute, getDisclaimerPath, renderDisclaimerPage } from './disclaimer'
import poisRoute from './pois'
import { getSprungbrettExtraPath, renderSprungbrettPage, SPRUNGBRETT_ROUTE, sprungbrettRoute } from './sprungbrett'
import { getWohnenExtraPath, renderWohnenPage, WOHNEN_ROUTE, wohnenRoute } from './wohnen'
import { getSearchPath, renderSearchPage, SEARCH_ROUTE, searchRoute } from './search'
import { MAIN_DISCLAIMER_ROUTE, mainDisclaimerRoute, renderMainDisclaimerPage } from './mainDisclaimer'
import { I18N_REDIRECT_ROUTE, i18nRedirectRoute, renderI18nPage } from './i18nRedirect'
import { LANDING_ROUTE, landingRoute, renderLandingPage } from './landing'
import type { Route as RouterRouteType } from 'redux-first-router'
import Route from './Route'

export const requiredPayloads = {
  [EXTRAS_ROUTE]: (payloads: AllPayloadsType) =>
    ({extras: payloads.extrasPayload, cities: payloads.citiesPayload}),
  [DISCLAIMER_ROUTE]: (payloads: AllPayloadsType) =>
    ({disclaimer: payloads.disclaimerPayload, cities: payloads.citiesPayload}),
  [SPRUNGBRETT_ROUTE]: (payloads: AllPayloadsType) => ({
    extras: payloads.extrasPayload, cities: payloads.citiesPayload, sprungbrettJobs: payloads.sprungbrettJobsPayload}),
  [WOHNEN_ROUTE]: (payloads: AllPayloadsType) =>
    ({extras: payloads.extrasPayload, cities: payloads.citiesPayload, offers: payloads.wohnenPayload}),
  [SEARCH_ROUTE]: (payloads: AllPayloadsType) =>
    ({categories: payloads.categoriesPayload, cities: payloads.citiesPayload}),
  [MAIN_DISCLAIMER_ROUTE]: () => ({}),
  [I18N_REDIRECT_ROUTE]: (payloads: AllPayloadsType) => ({cities: payloads.citiesPayload}),
  [LANDING_ROUTE]: (payloads: AllPayloadsType) => ({cities: payloads.citiesPayload})
}

// this maps all goToRoute actions to the right routes (except from the NOT_FOUND route)
// the order is important, routes declared first are served first, so i.e. if you put the mainDisclaimer route after
// the i18nRedirect route, "/disclaimer" also matches the i18nRedirect route
export const routesMap: {[string]: RouterRouteType} = {
  [MAIN_DISCLAIMER_ROUTE]: mainDisclaimerRoute,
  [I18N_REDIRECT_ROUTE]: i18nRedirectRoute,
  [LANDING_ROUTE]: landingRoute,
  [eventsRoute.name]: eventsRoute,
  [SPRUNGBRETT_ROUTE]: sprungbrettRoute,
  [WOHNEN_ROUTE]: wohnenRoute,
  [EXTRAS_ROUTE]: extrasRoute,
  [DISCLAIMER_ROUTE]: disclaimerRoute,
  [SEARCH_ROUTE]: searchRoute,
  [poisRoute.name]: poisRoute,
  [categoriesRoute.name]: categoriesRoute
}

const routes: {[string]: Route} = {
  [categoriesRoute.name]: categoriesRoute,
  [eventsRoute.name]: eventsRoute,
  [EXTRAS_ROUTE]: renderExtrasPage,
  [DISCLAIMER_ROUTE]: renderDisclaimerPage,
  [poisRoute.name]: poisRoute,
  [SPRUNGBRETT_ROUTE]: renderSprungbrettPage,
  [WOHNEN_ROUTE]: renderWohnenPage,
  [SEARCH_ROUTE]: renderSearchPage,
  [MAIN_DISCLAIMER_ROUTE]: renderMainDisclaimerPage,
  [I18N_REDIRECT_ROUTE]: renderI18nPage,
  [LANDING_ROUTE]: renderLandingPage
}

export default routes
