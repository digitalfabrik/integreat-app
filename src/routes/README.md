This package contains all the routes. Routes can contain containers, components, hocs, actions and reducers
