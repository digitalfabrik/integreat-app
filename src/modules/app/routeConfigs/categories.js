// @flow

import RouteConfig from './RouteConfig'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import categoriesEndpoint from '../../endpoint/endpoints/categories'
import type { AllPayloadsType, GetLanguageChangePathParamsType, GetPageTitleParamsType } from './RouteConfig'
import CategoriesMapModel from '../../endpoint/models/CategoriesMapModel'
import Payload from '../../endpoint/Payload'
import CityModel from '../../endpoint/models/CityModel'

export type CategoriesRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|categories: Payload<CategoriesMapModel>, cities: Payload<Array<CityModel>>|}

export const CATEGORIES_ROUTE = 'CATEGORIES'

const getCategoriesPath = ({city, language}: CategoriesRouteParamsType): string => `/${city}/${language}`

/**
 * CategoriesRoute, matches /augsburg/de*
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const categoriesRoute: Route = {
  path: '/:city/:language/:categoryPath*',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()
    const { city, language } = state.location.payload

    await fetchData(categoriesEndpoint, dispatch, state.categories, { city, language })
  }
}

const getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType =>
  ({categories: payloads.categoriesPayload, cities: payloads.citiesPayload})

const getLanguageChangePath = ({location, categories, language}: GetLanguageChangePathParamsType) => {
  const {city} = location.payload
  if (categories) {
    const category = categories.findCategoryByPath(location.pathname)
    if (category && category.id !== 0) {
      return category.availableLanguages.get(language) || null
    }
  }
  return getCategoriesPath({city, language})
}

const getPageTitle = ({t, categories, cityName, pathname}: GetPageTitleParamsType) => {
  const category = categories && categories.findCategoryByPath(pathname)
  return `${category && !category.isRoot() ? `${category.title} - ` : ''}${cityName}`
}

class CategoriesRouteConfig extends RouteConfig<CategoriesRouteParamsType, RequiredPayloadsType> {
  constructor () {
    super({
      name: CATEGORIES_ROUTE,
      getRequiredPayloads,
      route: categoriesRoute,
      getRoutePath: getCategoriesPath,
      getLanguageChangePath,
      getPageTitle
    })
  }
}

export default CategoriesRouteConfig
