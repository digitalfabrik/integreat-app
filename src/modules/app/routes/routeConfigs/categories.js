// @flow

import type { AllPayloadsType, GetLanguageChangePathParamsType, GetPageTitleParamsType } from '../types'
import Payload from '../../../endpoint/Payload'
import CityModel from '../../../endpoint/models/CityModel'
import CategoriesMapModel from '../../../endpoint/models/CategoriesMapModel'
import RouteConfig from './RouteConfig'
import React from 'react'
import CategoriesPage from '../../../../routes/categories/containers/CategoriesPage'
import { getCategoriesPath } from '../categories'

type RequiredPayloadType = {|categories: Payload<CategoriesMapModel>, cities: Payload<Array<CityModel>>|}

const renderCategoriesPage = ({ categories, cities }: RequiredPayloadType) =>
  <CategoriesPage categories={categories.data} cities={cities.data} />

const getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadType =>
  ({ categories: payloads.categoriesPayload, cities: payloads.citiesPayload })

const getLanguageChangePath = ({location, categories, language}: GetLanguageChangePathParamsType) => {
  const {city} = location.payload
  if (categories) {
    const category = categories.findCategoryByPath(location.pathname)
    if (category && category.id !== 0) {
      return category.availableLanguages.get(language) || null
    }
  }
  return getCategoriesPath({city, language})
}

const getPageTitle = ({t, categories, cityName, pathname}: GetPageTitleParamsType) => {
  const category = categories && categories.findCategoryByPath(pathname)
  return `${category && !category.isRoot() ? `${category.title} - ` : ''}${cityName}`
}

const categoriesRouteConfig: RouteConfig<RequiredPayloadType> = new RouteConfig({
  renderPage: renderCategoriesPage,
  getRequiredPayloads,
  getLanguageChangePath,
  getPageTitle
})

export default categoriesRouteConfig
