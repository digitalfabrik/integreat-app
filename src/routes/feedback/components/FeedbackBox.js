// @flow

import * as React from 'react'
import 'react-dropdown/style.css'

import CityModel from '../../../modules/endpoint/models/CityModel'
import { translate } from 'react-i18next'
import styled from 'styled-components'
import FeedbackEndpoint, {
  CATEGORIES_FEEDBACK_TYPE,
  DEFAULT_FEEDBACK_LANGUAGE,
  EXTRA_FEEDBACK_TYPE,
  INTEGREAT_INSTANCE,
  PAGE_FEEDBACK_TYPE,
  SEARCH_FEEDBACK_TYPE
}
  from '../../../modules/endpoint/FeedbackEndpoint'
import type { TFunction } from 'react-i18next'
import CleanLink from '../../../modules/common/components/CleanLink'
import type { FeedbackDataType } from '../../../modules/endpoint/FeedbackEndpoint'
import { CATEGORIES_ROUTE } from '../../../modules/app/routes/categories'
import { EVENTS_ROUTE } from '../../../modules/app/routes/events'
import { EXTRAS_ROUTE } from '../../../modules/app/routes/extras'
import { SEARCH_ROUTE } from '../../../modules/app/routes/search'
import { DISCLAIMER_ROUTE } from '../../../modules/app/routes/disclaimer'
import FeedbackHeader from './FeedbackHeader'
import FeedbackComment from './FeedbackComment'
import type { LocationState } from 'redux-first-router'
import { goToFeedback } from '../../../modules/app/routes/feedback'
import FeedbackDropdownItem from '../FeedbackDropdownItem'

const StyledFeedbackBox = styled.div`
  display: flex;
  width: 400px;
  height: auto;
  box-sizing: border-box;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px;
  border-radius: 10px;
  border-color: #585858;
  font-size: ${props => props.theme.fonts.contentFontSize};
`

export const Description = styled.div`
  padding: 10px 0 5px;
`

const SubmitButton = styled(CleanLink)`
  margin: 15px 0;
  padding: 5px;
  background-color: ${props => props.theme.colors.themeColor};
  color: ${props => props.theme.colors.backgroundAccentColor};
  text-align: center;
  border-radius: 0.25em;
`

type PropsType = {
  cities: Array<CityModel>,
  title?: string,
  alias?: string,
  id?: number,
  query?: string,
  isPositiveRatingSelected: boolean,
  location: LocationState,
  isOpen: boolean,
  commentMessageOverride: ?string,
  hideHeader: boolean,
  t: TFunction
}

type StateType = {
  feedbackOptions: Array<FeedbackDropdownItem>,
  selectedFeedbackOption: FeedbackDropdownItem,
  comment: string
}

export class FeedbackBox extends React.Component<PropsType, StateType> {
  static defaultProps = {
    hideHeader: false
  }
  constructor (props: PropsType) {
    super(props)
    const feedbackOptions = this.getFeedbackOptions()
    this.state = {feedbackOptions: feedbackOptions, selectedFeedbackOption: feedbackOptions[0], comment: ''}
  }

  onCommentChanged = (event: SyntheticInputEvent<HTMLTextAreaElement>) => this.setState({comment: event.target.value})

  onFeedbackOptionChanged = (selectedDropdown: FeedbackDropdownItem) => {
    this.setState({selectedFeedbackOption: selectedDropdown})
  }

  componentDidUpdate (prevProps: PropsType) {
    const {isOpen} = this.props
    const prevIsOpen = prevProps.isOpen

    // If the FeedbackBox is opened, we have to reset and initialize the state
    if (prevIsOpen !== isOpen && isOpen) {
      /* eslint-disable react/no-did-update-set-state */
      const feedbackOptions = this.getFeedbackOptions()
      const selectedFeedbackOption = feedbackOptions[0]
      this.setState({feedbackOptions: feedbackOptions, selectedFeedbackOption: feedbackOptions[0], comment: ''})
      FeedbackEndpoint.postData(this.getFeedbackData(selectedFeedbackOption, ''))
    }
  }

  getFeedbackData = (selectedFeedbackOption: FeedbackDropdownItem, comment: string): FeedbackDataType => {
    const {location, query, isPositiveRatingSelected, id, alias} = this.props
    const {city, language} = location.payload
    return {
      feedbackType: selectedFeedbackOption.feedbackType,
      isPositiveRating: isPositiveRatingSelected,
      comment,
      id,
      city: city || INTEGREAT_INSTANCE,
      language: language || DEFAULT_FEEDBACK_LANGUAGE,
      alias,
      query
    }
  }

  onSubmit = () => {
    const {selectedFeedbackOption, comment} = this.state
    FeedbackEndpoint.postData(this.getFeedbackData(selectedFeedbackOption, comment))
  }

  getFeedbackOptions = (): Array<FeedbackDropdownItem> => {
    const {cities, location, t} = this.props
    const {city} = location.payload

    const options = []
    const currentPageFeedbackLabel = this.getCurrentPageFeedbackLabel()
    if (currentPageFeedbackLabel) {
      options.push(this.getFeedbackOption(currentPageFeedbackLabel, this.getCurrentPageFeedbackType()))
    }
    if (city) {
      const cityTitle = CityModel.findCityName(cities, city)
      options.push(this.getFeedbackOption(`${t('contentOfCity')} ${cityTitle}`, CATEGORIES_FEEDBACK_TYPE))
    }

    options.push(this.getFeedbackOption(t('technicalTopic'), CATEGORIES_FEEDBACK_TYPE))

    return options
  }

  getFeedbackOption = (label: string, feedbackType: ?string): FeedbackDropdownItem =>
    new FeedbackDropdownItem(label, feedbackType)

  getCurrentPageFeedbackLabel = (): ?string => {
    const {location, id, alias, title, query, t} = this.props
    const type = location.type

    if (type === CATEGORIES_ROUTE && id && title) {
      return `${t('contentOfPage')} '${title}'`
    } else if (type === EVENTS_ROUTE && id && title) {
      return `${t('news')} '${title}'`
    } else if (type === EXTRAS_ROUTE && alias && title) {
      return `${t('extra')} '${title}'`
    } else if (type === SEARCH_ROUTE && query) {
      return `${t('searchFor')} '${query}'`
    } else if (type === DISCLAIMER_ROUTE) {
      return `${t('disclaimer')}`
    } else {
      return null
    }
  }

  getCurrentPageFeedbackType = (): string | null => {
    const {type} = this.props.location
    if (type === SEARCH_ROUTE) {
      return SEARCH_FEEDBACK_TYPE
    } else if (type === EXTRAS_ROUTE) {
      return EXTRA_FEEDBACK_TYPE
    } else {
      return PAGE_FEEDBACK_TYPE
    }
  }

  render () {
    const {selectedFeedbackOption, feedbackOptions, comment} = this.state
    const {t, isPositiveRatingSelected, location, commentMessageOverride, hideHeader} = this.props
    return (
      <StyledFeedbackBox>
        {!hideHeader && (
          <FeedbackHeader
            location={location}
            selectedFeedbackOption={selectedFeedbackOption}
            feedbackOptions={feedbackOptions}
            onFeedbackOptionChanged={this.onFeedbackOptionChanged} />
        )}
        <FeedbackComment
          comment={comment}
          commentMessageOverride={commentMessageOverride}
          isPositiveRatingSelected={isPositiveRatingSelected}
          onCommentChanged={this.onCommentChanged} />
        <SubmitButton to={goToFeedback(location)} onClick={this.onSubmit}>
          {t('send')}
        </SubmitButton>
      </StyledFeedbackBox>
    )
  }
}

export default translate('feedback')(FeedbackBox)
