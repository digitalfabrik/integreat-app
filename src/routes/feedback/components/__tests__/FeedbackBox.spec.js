import React from 'react'
import { shallow } from 'enzyme'

import { CATEGORIES_ROUTE } from '../../../../modules/app/routes/categories'
import CityModel from '../../../../modules/endpoint/models/CityModel'
import { FeedbackBox } from '../FeedbackBox'

describe('FeedbackBox', () => {
  const location = {type: CATEGORIES_ROUTE, payload: {city: 'augsburg', language: 'de'}}
  const cities = [
    new CityModel({
      name: 'Augsburg',
      code: 'augsburg',
      live: true,
      eventsEnabled: true,
      extrasEnabled: false
    })
  ]
  const t = key => key

  it('should render a FeedbackBox with Header', () => {
    const component = shallow(
      <FeedbackBox
        query={'ab'}
        location={location}
        cities={cities}
        isPositiveRatingSelected={false}
        isOpen
        t={t} />
    )
    expect(component).toMatchSnapshot()
  })

  it('should render a FeedbackBox without Header', () => {
    const component = shallow(
      <FeedbackBox
        query={'ab'}
        location={location}
        cities={cities}
        isPositiveRatingSelected={false}
        isOpen
        t={t}
        hideHeader />
    )
    expect(component).toMatchSnapshot()
  })
})
