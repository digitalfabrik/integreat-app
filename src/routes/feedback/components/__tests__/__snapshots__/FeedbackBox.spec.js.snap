// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackBox should render a FeedbackBox with Header 1`] = `
<FeedbackBox__StyledFeedbackBox>
  <Translate(FeedbackHeader)
    feedbackOptions={
      Array [
        FeedbackDropdownItem {
          "feedbackType": "categories",
          "label": "contentOfCity Augsburg",
          "value": "contentOfCity Augsburg",
        },
        FeedbackDropdownItem {
          "feedbackType": "categories",
          "label": "technicalTopic",
          "value": "technicalTopic",
        },
      ]
    }
    location={
      Object {
        "payload": Object {
          "city": "augsburg",
          "language": "de",
        },
        "type": "CATEGORIES",
      }
    }
    onFeedbackOptionChanged={[Function]}
    selectedFeedbackOption={
      FeedbackDropdownItem {
        "feedbackType": "categories",
        "label": "contentOfCity Augsburg",
        "value": "contentOfCity Augsburg",
      }
    }
  />
  <Translate(FeedbackComment)
    comment=""
    isPositiveRatingSelected={false}
    onCommentChanged={[Function]}
  />
  <FeedbackBox__SubmitButton
    onClick={[Function]}
    to={
      Object {
        "payload": Object {
          "city": "augsburg",
          "language": "de",
        },
        "query": undefined,
        "type": "CATEGORIES",
      }
    }
  >
    send
  </FeedbackBox__SubmitButton>
</FeedbackBox__StyledFeedbackBox>
`;

exports[`FeedbackBox should render a FeedbackBox without Header 1`] = `
<FeedbackBox__StyledFeedbackBox>
  <Translate(FeedbackComment)
    comment=""
    isPositiveRatingSelected={false}
    onCommentChanged={[Function]}
  />
  <FeedbackBox__SubmitButton
    onClick={[Function]}
    to={
      Object {
        "payload": Object {
          "city": "augsburg",
          "language": "de",
        },
        "query": undefined,
        "type": "CATEGORIES",
      }
    }
  >
    send
  </FeedbackBox__SubmitButton>
</FeedbackBox__StyledFeedbackBox>
`;
